### PR TITLE
Fix backpropagation bugs in SO3 and SE3 log_map

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: theseus/geometry/se3.py [hunk 1]
```diff
@@ -254,13 +254,12 @@ def _log_map_impl(self) -> torch.Tensor:
         NEAR_PI_EPS = 1e-7
         NEAR_ZERO_EPS = 5e-3
 
-        ret = torch.zeros(self.shape[0], 6, dtype=self.dtype, device=self.device)
-
-        ret[:, 3] = 0.5 * (self[:, 2, 1] - self[:, 1, 2])
-        ret[:, 4] = 0.5 * (self[:, 0, 2] - self[:, 2, 0])
-        ret[:, 5] = 0.5 * (self[:, 1, 0] - self[:, 0, 1])
+        sine_axis = torch.zeros(self.shape[0], 3, dtype=self.dtype, device=self.device)
```
Comment (mhmukadam) [inline theseus/geometry/se3.py:257]: I am confused about the difference between #108 and #109.

File: theseus/geometry/so3.py [hunk 2]
```diff
@@ -185,12 +185,12 @@ def exp_map(tangent_vector: torch.Tensor) -> "SO3":
         return ret
 
     def _log_map_impl(self) -> torch.Tensor:
-        ret = torch.zeros(self.shape[0], 3, dtype=self.dtype, device=self.device)
-        ret[:, 0] = 0.5 * (self[:, 2, 1] - self[:, 1, 2])
-        ret[:, 1] = 0.5 * (self[:, 0, 2] - self[:, 2, 0])
-        ret[:, 2] = 0.5 * (self[:, 1, 0] - self[:, 0, 1])
+        sine_axis = torch.zeros(self.shape[0], 3, dtype=self.dtype, device=self.device)
```
Comment (mhmukadam) [inline theseus/geometry/so3.py:188]: We should also add unit tests that will fail with the previous version (something that checks the gradient).

File: theseus/geometry/se3.py (hunk 1 above)
Comment (mhmukadam) [reply theseus/geometry/se3.py:257]: 109 supersedes 108.

File: theseus/geometry/so3.py (hunk 2 above)
Comment (mhmukadam) [reply theseus/geometry/so3.py:188]: Will be added in another PR.

**Review by mhmukadam — APPROVED** [review]

**Review by luisenp — APPROVED** [review]
LGTM! Feel free to merge.
